tests/roles/run_tests/tasks/main.yml

```diff
@@ -45,27 +45,13 @@
   include_tasks: move_back.yml
   when: v1aX_integration_test_action in repivot_actions
 
-- name: Deprovision worker nodes
-  kubernetes.core.k8s:
-    state: absent
-    src: "{{ TEMP_GEN_DIR }}/{{ CAPM3_VERSION }}_workers_{{ IMAGE_OS }}.yaml"
-    namespace: "{{ NAMESPACE }}"
-  ignore_errors: yes
-  when: v1aX_integration_test_action in deprovision_workers_actions
-
-- name: Deprovision control plane
-  kubernetes.core.k8s:
-    state: absent
-    src: "{{ TEMP_GEN_DIR }}/{{ CAPM3_VERSION }}_controlplane_{{ IMAGE_OS }}.yaml"
-    namespace: "{{ NAMESPACE }}"
-  ignore_errors: yes
-  when: v1aX_integration_test_action in deprovision_controlplane_actions
-
 - name: Deprovision cluster
   kubernetes.core.k8s:
     state: absent
-    src: "{{ TEMP_GEN_DIR }}/{{ CAPM3_VERSION }}_cluster_{{ IMAGE_OS }}.yaml"
```

Comment:
Not sure this was used to delete also the IPPools ? are they deleted automatically when deleting the cluster

Reply:
They should be deleted when the cluster is deleted because they have an ownerRef to it.